### PR TITLE
[Frontend] Validation for Invite Admin Form  #510

### PR DESCRIPTION
**Pull request by bishakhne0gi**
## [Frontend] Validation for Invite Admin Form 

 Closes: #510

### CODE REVIEW

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
@bishakhne0gi 
please do the following -
1. Covert the `div` in `form`
2. You need to all the validation function `onChange` and the `onSubmit` function on the form submission.

**Comment by bishakhne0gi** [comment]
Ok, i am doing that.

**Review by bishakhne0gi — COMMENTED** [review]
I have done the required changes and its working properly. Please check it once

File: frontend/package.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@hookform/resolvers": "^1.3.5",
```
Comment (Kajol-Kumari) [inline frontend/package.json:6]: why do we need this?

File: frontend/package.json [hunk 2]
```diff
@@ -12,12 +13,14 @@
...
     "bootstrap-social": "^5.1.1",
     "font-awesome": "^4.7.0",
+    "joi-browser": "^13.4.0",
     "mdbreact": "^5.0.1",
     "node-sass": "^4.14.1",
     "prop-types": "^15.7.2",
     "react": "^17.0.1",
     "react-card-flip": "^1.0.11",
     "react-dom": "^17.0.1",
+    "react-hook-form": "^6.15.4",
```
Comment (Kajol-Kumari) [inline frontend/package.json:23]: again, why do we need this?

File: frontend/package.json [hunk 3]
```diff
@@ -33,7 +36,8 @@
     "semantic-ui-react": "^2.0.1",
     "styled-components": "^5.2.1",
     "suneditor-react": "^2.15.2",
-    "web-vitals": "^0.2.4"
+    "web-vitals": "^0.2.4",
+    "yup": "^0.32.9"
```
Comment (Kajol-Kumari) [inline frontend/package.json:40]: again, why is this needed?

File: frontend/src/pages/Admin/Components/Setting/Invite/Invite.jsx [hunk 4]
```diff
@@ -1,8 +1,67 @@
...
 import style from "./Invite.module.scss";
+import Joi from "joi-browser";
+export function Invite() {
+  const [formdata, setFormData] = useState({
+    name: "",
+    email: "",
+  });
+  const [formerrors, setFormErrors] = useState({});
+  const schema = {
+    name: Joi.string().required(),
```
Comment (Kajol-Kumari) [inline frontend/src/pages/Admin/Components/Setting/Invite/Invite.jsx:12]: we don't have any field that takes `name`. So,  please remove this from the schema

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Hey @bishakhne0gi please remove `package-lock.json` file from this PR and make the required changes

**Review by bishakhne0gi — COMMENTED** [review]
The paclkage.json is removed with other mentioned changes

**Comment by Kajol-Kumari** [comment]
> The package-lock.json is removed with other mentioned changes

Hey @bishakhne0gi I can still see `package-lock.json` file in this PR. Can you please review this

**Comment by bishakhne0gi** [comment]
In the updated with required changes commit i deleted it.

**Comment by bishakhne0gi** [comment]
Can you please mention which package-lock.json.

**Comment by Kajol-Kumari** [comment]
> Can you please mention which package-lock.json.

Please don't delete it, instead revert it's changes using `git checkout HEAD -- <my-file.txt>`

**Comment by bishakhne0gi** [comment]
@Kajol-Kumari It is not working, i tried.
Can you please help?

**Comment by bishakhne0gi** [comment]
Is it okay now?

File: frontend/package.json [hunk 5]
```diff
@@ -32,8 +33,7 @@
     "semantic-ui-css": "^2.4.1",
     "semantic-ui-react": "^2.0.1",
     "styled-components": "^5.2.1",
-    "suneditor-react": "^2.15.2",
-    "web-vitals": "^0.2.4"
```
Comment (Kajol-Kumari) [inline frontend/package.json:36]: why this package has been removed?

File: frontend/src/pages/Admin/Components/Setting/Invite/Invite.jsx [hunk 6]
```diff
@@ -1,8 +1,66 @@
-import React from "react";
+import React, { useState } from "react";
 import { Button2 } from "../../../../../components/util/Button";
 import style from "./Invite.module.scss";
+import Joi from "joi-browser";
+export function Invite() {
+  const [formdata, setFormData] = useState({
+    name: "",
```
Comment (Kajol-Kumari) [inline frontend/src/pages/Admin/Components/Setting/Invite/Invite.jsx:7]: we don't need this as our form doesn't contain any `name` field

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Hey @bishakhne0gi please do the following changes Also I can see the default error msg as well, along with joi-validator msg, so please remove the default error msg 

error_msg_bug

**Comment by bishakhne0gi** [comment]
> Hey @bishakhne0gi please do the following changes Also I can see the default error msg as well, along with joi-validator msg, so please remove the default error msg
> 
> error_msg_bug

Can you please tell how to remove because find the default message part in my code?

File: frontend/package.json (hunk 5 above)
Comment (bishakhne0gi) [reply frontend/package.json:36]: > why this package has been removed?

Yes, i am installing it again. I didnt remove but somehow it got reoved. I am doing it

**Comment by bishakhne0gi** [comment]
The default message is shown in Resource Form also. Check it once

**Comment by Kajol-Kumari** [comment]
> > Hey @bishakhne0gi please do the following changes Also I can see the default error msg as well, along with joi-validator msg, so please remove the default error msg
> > error_msg_bug
> 
> Can you please tell how to remove because cannot find the default message part in my code?

IMO the `joi-validator` isn't implemented correctly. I recommend you to check it as I can see a lot of redundant code in there. You can see how it's being implemented in other component for your reference

**Comment by Kajol-Kumari** [comment]
> The default message is shown in Resource Form Validation also. Check it once

No, I can't find it. Can you please point out the field name?

**Comment by bishakhne0gi** [comment]
Resources field in the website. It too has email form validation part. In the live server I used it, it gave me joi-browser validation and the default message error too.

**Comment by Kajol-Kumari** [comment]
> Resources field in the website. It too has email form validation part. In the live server I used it, it gave me joi-browser validation and the default message error too.
> 
> https://hitk-tech-community.netlify.app/resources

@bishakhne0gi please correct me if I am missing here something?
resource_form

**Comment by bishakhne0gi** [comment]
@Kajol-Kumari can you please excuse me for two days? As i'm out of station and have very poor internet connection.

**Comment by bishakhne0gi** [comment]
> > Resources field in the website. It too has email form validation part. In the live server I used it, it gave me joi-browser validation and the default message error too.
> > https://hitk-tech-community.netlify.app/resources
> 
> @bishakhne0gi please correct me if I am missing here something?
> resource_form

Yes, I am telling after writing the email press enter not enter, the default error will come. Is there any provision to send ss then i can attach the ss here.

**Review by Kajol-Kumari — APPROVED** [review]
Thanks for fixing everything @bishakhne0gi 
LGMT :+1:

**Comment by bishakhne0gi** [comment]
It was a very nice experience to work with you. Thank you for constantly guiding me. I want to work with you more and keep learning. Can you please tell me whether i will get the points?

**Comment by jackfrost13** [comment]
Hey @bishakhne0gi yes you will get the points, all merged pr's are eligible for the points. The points depends on the level of the issue and it is taken care entirely by the organisers, from our end we only assign levels to different issues. 
|Level|Points|
|-|-|
|Level0| 5|
|Level1 | 10|
|Level2| 15|
|Level3| 25|

**Comment by bishakhne0gi** [comment]
Thank you so much @jackfrost13 . Then as per the list, i shall be awarded with 15points right?

**Comment by jackfrost13** [comment]
Yes
